src/SourceTerms/Leakage/CPU_Src_Leakage_ComputeLeak.cpp

```diff
@@ -0,0 +1,1150 @@
+                                "10_chi_Ross_Nue", "11_chi_Ross_Nua", "12_chi_Ross_Nux",
+                                "13_HeatFlux_Nue", "14_HeatFlux_Nua", "15_dEdt", "16_dYedt" );
+         fprintf( File_Leakage, "#%13s %14s %16s %16s %16s %16s %16s %16s %16s %16s %16s %16s %16s %16s %16s %16s\n",
+                                "[cm]", "[1]", "[g/cm^3]", "[Kelvin]", "[dimensionless]", "[cm^2/s^2]",
+                                "[dimensionless]", "[dimensionless]", "[dimensionless]",
+                                "[dimensionless]", "[dimensionless]", "[dimensionless]",
+                                "[erg/cm^2/s]", "[erg/cm^2/s]", "[erg/g/s]", "[1/s]" );
+
+         for (int i=0; i<NRadius; i++)
+            fprintf( File_Leakage, "%14.6e %14ld %16.6e %16.6e %16.6e %16.6e %16.6e %16.6e %16.6e %16.6e %16.6e %16.6e %16.6e %16.6e %16.6e %16.6e\n",
+                                   0.5 * (Edge_CGS[i] + Edge_CGS[i+1]), NCell[TID][i],
+                                   Dens_CGS[TID][i], Temp_Kelv[TID][i], Ye[TID][i], Eint_Code[TID][i] / Dens_Code[TID][i] * SQR(UNIT_V),
+                                   tau_Ruff[i][j][0], tau_Ruff[i][j][1], tau_Ruff[i][j][2],
+                                   chi_Ross[i][j][0], chi_Ross[i][j][1], chi_Ross[i][j][2],
+                                   Heat_Flux[i][j][0] / Const_hc_MeVcm_CUBE, Heat_Flux[i][j][1] / Const_hc_MeVcm_CUBE, dEdt[TID][i], dYedt[TID][i] );
+
+         fclose( File_Leakage );
+#        endif // #ifdef GAMER_DEBUG
+      } // for (int j=IRay_start; j<IRay_end; j++)
+   } // #  pragma omp parallel for schedule( runtime )
+
+
+// wait for unfinished jobs because jobs could be not evenly distributed (NRay % MPI_NRank != 0)
+   MPI_Barrier( MPI_COMM_WORLD );
+
+
+// sum over all OpenMP threads
+   for (int t=0; t<NT;             t++) {
+   for (int k=0; k<NType_Neutrino; k++) {
+      EAve [k] += OMP_EAve [t][k];
+      RadNS[k] += OMP_RadNS[t][k];
+   }}
+
+
+// collect data from all ranks
+#  ifndef SERIAL
+   MPI_Allreduce( MPI_IN_PLACE, tau_Ruff,  NRadius*NRay*NType_Neutrino, MPI_GAMER_REAL, MPI_SUM, MPI_COMM_WORLD );
+   MPI_Allreduce( MPI_IN_PLACE, chi_Ross,  NRadius*NRay*NType_Neutrino, MPI_GAMER_REAL, MPI_SUM, MPI_COMM_WORLD );
+   MPI_Allreduce( MPI_IN_PLACE, Heat_Flux, NRadius*NRay*NType_Neutrino, MPI_GAMER_REAL, MPI_SUM, MPI_COMM_WORLD );
+   MPI_Allreduce( MPI_IN_PLACE, HeatE_Rms,         NRay*NType_Neutrino, MPI_GAMER_REAL, MPI_SUM, MPI_COMM_WORLD );
+   MPI_Allreduce( MPI_IN_PLACE, HeatE_Ave,         NRay*NType_Neutrino, MPI_GAMER_REAL, MPI_SUM, MPI_COMM_WORLD );
+   MPI_Allreduce( MPI_IN_PLACE, EAve,                   NType_Neutrino, MPI_DOUBLE,     MPI_SUM, MPI_COMM_WORLD );
+   MPI_Allreduce( MPI_IN_PLACE, RadNS,                  NType_Neutrino, MPI_DOUBLE,     MPI_SUM, MPI_COMM_WORLD );
+#  endif // ifndef SERIAL
+
+
+// store the radius of neutrino sphere and mean neutrino energy
+   for (int k=0; k<NType_Neutrino; k++)  {
```

Comment:
does neutrino luminosity also stored?

Reply:
The luminosity `Heat_Flux` is updated directly [here](https://github.com/hfhsieh/gamer-fork/blob/ccsn_leakage/src/SourceTerms/Leakage/CPU_Src_Leakage_ComputeLeak.cpp#L623). The code here is to compute the average of neutrino sphere and energy among all the rays.